corehq/apps/products/forms.py

```diff
@@ -22,7 +22,7 @@ def clean(self, value):
 
 class ProductForm(forms.Form):
     name = forms.CharField(max_length=100)
-    code = forms.CharField(label=ugettext_noop("Product ID"), max_length=10, required=False)
+    code = forms.CharField(label=ugettext_noop("Product ID"), max_length=100, required=False)
```

Comment:
@esoergel Not sure, but yeah keeping sms reporting messages short might be one likely reason.  I suppose if the db allows for 100 chars there's no harm in letting you enter that many in the ui, and anyone who uses sms to report would just have to keep them short to keep the sms reporting manageable.